include/boost/thread/concurrent_queues/sync_timed_queue.hpp

```diff
@@ -122,8 +122,7 @@ namespace detail
     inline bool not_empty_and_time_reached(lock_guard<mutex>& lk) const;
 
     bool wait_to_pull(unique_lock<mutex>&);
-    template <class WClock, class Duration>
-    queue_op_status wait_to_pull_until(unique_lock<mutex>&, chrono::time_point<WClock, Duration> const& tp);
+    queue_op_status wait_to_pull_until(unique_lock<mutex>&, TimePoint const& tp);
```

Comment:
Why do you want to be more generic here? Do you have other timepoints other than boost::chrono::timepoint? Are you thinking on std::chrono::timepoint?

Reply:
Since `wait_to_pull_until()` is an internal/private function, it simplified the code to have the type of `tp` match the type of `super::data_.top().time`, which is also of type `TimePoint`. Then in `pull_until()` I use `time_point_cast()` to convert from the user's time point type to `TimePoint`.

Reply:
Okay, I see.